### ISSUE TITLE
Semantic Tokens are not requested from installed extension, while it does work when running "2nd level"

### ISSUE THREAD

**Issue by DavyLandman** [microsoft/vscode#118639]
Issue Type: <b>Bug</b>

We are working on a LSP implementation for Rascal and are really happy LSP & vscode now support Semantic Tokens.

Currently our LSP implementation works quite nicely when running as 2nd level (starting vscode from inside vscode). Outline, definitions and semantic tokens work.

After packaging a vsix file and installing it in "1st level", outline & definitions still work, but semantic tokens are never requested from the client side.

Our server is reporting the `SemanticTokensWithRegistrationOptions` in the same way as for 2nd level. I cannot find any logs on the vs-code client side that can help me along.

Is there some extra set of debug tasks I could setup to figure out what's going on? Are there specific conditions that the Semantic Tokens feature needs that change when not running 2nd level?

VS Code version: Code 1.54.1 (f30a9b73e8ffc278e71575118b6bf568f04587c8, 2021-03-04T22:38:31.419Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2904)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.82GB (6.94GB free)|
|Process Argv|--crash-reporter-id 2bad8817-c94f-41f1-8efe-a70e5e999151|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (53)</summary>

Extension|Author (truncated)|Version
---|---|---
asciidoctor-vscode|asc|2.8.7
language-hugo-vscode|bud|1.3.0
better-toml|bun|0.3.2
gitignore|cod|0.6.0
vscode-markdownlint|Dav|0.39.0
vscode-eslint|dba|2.1.16
githistory|don|0.6.15
xml|Dot|2.5.1
gitlens|eam|11.3.0
tsl-problem-matcher|eam|0.4.0
vscode-html-css|ecm|1.10.2
EditorConfig|Edi|0.16.4
scad|efb|1.0.1
tslint|eg2|1.0.45
vscode-npm-script|eg2|0.3.16
lex-flex-yacc-bison|fau|0.0.3
git-project-manager|fel|1.7.1
go|gol|0.23.0
xtext-lang|gra|0.4.0
gitattributes|has|0.4.1
latex-workshop|Jam|8.16.1
plantuml|jeb|2.14.3
asciidoctor-vscode|joa|2.8.0
vscode-map-preview|jum|0.5.8
rainbow-csv|mec|1.8.1
postcss-language|mhm|2.1.0
vscode-docker|ms-|1.10.0
python|ms-|2021.2.636928669
jupyter|ms-|2021.3.619093157
remote-wsl|ms-|0.54.2
cpptools|ms-|1.2.2
sqltools|mtx|0.23.0
printcode|nob|3.0.0
material-icon-theme|PKi|4.5.0
platformio-ide|pla|2.3.0
fabric8-analytics|red|0.3.2
java|red|0.76.0
vscode-commons|red|0.0.4
LiveServer|rit|5.6.1
scala|sca|0.5.2
sonarlint-vscode|Son|1.20.1
code-spell-checker|str|1.10.2
shellcheck|tim|0.13.2
rascalmpl|use|0.0.1-SNAPSHOT
vscodeintellicode|Vis|1.2.11
vscode-java-debug|vsc|0.31.0
vscode-java-dependency|vsc|0.18.1
vscode-java-pack|vsc|0.13.0
vscode-java-test|vsc|0.28.1
vscode-maven|vsc|0.28.0
vscode-icons|vsc|11.1.0
vim|vsc|1.18.9
html-css-class-completion|Zig|1.20.0

(6 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspyt653:30253241
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstry914:30256638
pythonvsdeb440:30248342
pythonvsded773:30248341
pythonvspyt875:30259475
dockersubsetcf:30265999
pythontbcf:30265426
vspre833cf:30267465

```

</details>

**Comment by dbaeumer** [comment]
There is actually nothing special. When you package the extension do you webpack the code? If so does the same problem occur if you create the extension without webpack?

You can also enable tracing using `name.trace.server` setting where name conforms to the name you pass to the LanguageClient. The trace will then appear in the corresponding output channel.

**Comment by DavyLandman** [comment]
Our server implementation is a jar, and the exact same jar is run inside level 1 and level 2.

The [extension itself](https://github.com/cwi-swat/rascal-vscode/blob/master/rascal-vscode-extension/src/extension.ts) is indeed packaged by webpack, but all the other features are working, so I'm a bit confused on what could be happening there, as we don't do anything specific for semantic tokens. (also, do you have a small pointer on how to make it work without webpack?).


I have here the trace logs (i've removed the corresponding trace messages from our server).

# Level 1
```
[Trace - 21:18:21] Sending request 'initialize - (0)'.
Params: {
    "processId": 42664,
    "clientInfo": {
        "name": "Visual Studio Code",
        "version": "1.54.1"
    },
    "locale": "en-gb",
    "rootPath": "c:\\Users\\Davy\\swat.engineering\\rascal\\rascal-vscode\\rascal-lsp",
    "rootUri": "file:///c%3A/Users/Davy/swat.engineering/rascal/rascal-vscode/rascal-lsp",
    "capabilities": {
        "workspace": {
            "applyEdit": true,
            "workspaceEdit": {
                "documentChanges": true,
                "resourceOperations": [
                    "create",
                    "rename",
                    "delete"
                ],
                "failureHandling": "textOnlyTransactional",
                "normalizesLineEndings": true,
                "changeAnnotationSupport": {
                    "groupsOnLabel": true
                }
            },
            "didChangeConfiguration": {
                "dynamicRegistration": true
            },
            "didChangeWatchedFiles": {
                "dynamicRegistration": true
            },
            "symbol": {
                "dynamicRegistration": true,
                "symbolKind": {
                    "valueSet": [
                        1,
                        2,
                        3,
                        4,
                        5,
                        6,
                        7,
                        8,
                        9,
                        10,
                        11,
                        12,
                        13,
                        14,
                        15,
                        16,
                        17,
                        18,
                        19,
                        20,
                        21,
                        22,
                        23,
                        24,
                        25,
                        26
                    ]
                },
                "tagSupport": {
                    "valueSet": [
                        1
                    ]
                }
            },
            "codeLens": {
                "refreshSupport": true
            },
            "executeCommand": {
                "dynamicRegistration": true
            },
            "configuration": true,
            "workspaceFolders": true,
            "semanticTokens": {
                "refreshSupport": true
            },
            "fileOperations": {
                "dynamicRegistration": true,
                "didCreate": true,
                "didRename": true,
                "didDelete": true,
                "willCreate": true,
                "willRename": true,
                "willDelete": true
            }
        },
        "textDocument": {
            "publishDiagnostics": {
                "relatedInformation": true,
                "versionSupport": false,
                "tagSupport": {
                    "valueSet": [
                        1,
                        2
                    ]
                },
                "codeDescriptionSupport": true,
                "dataSupport": true
            },
            "synchronization": {
                "dynamicRegistration": true,
                "willSave": true,
                "willSaveWaitUntil": true,
                "didSave": true
            },
            "completion": {
                "dynamicRegistration": true,
                "contextSupport": true,
                "completionItem": {
                    "snippetSupport": true,
                    "commitCharactersSupport": true,
                    "documentationFormat": [
                        "markdown",
                        "plaintext"
                    ],
                    "deprecatedSupport": true,
                    "preselectSupport": true,
                    "tagSupport": {
                        "valueSet": [
                            1
                        ]
                    },
                    "insertReplaceSupport": true,
                    "resolveSupport": {
                        "properties": [
                            "documentation",
                            "detail",
                            "additionalTextEdits"
                        ]
                    },
                    "insertTextModeSupport": {
                        "valueSet": [
                            1,
                            2
                        ]
                    }
                },
                "completionItemKind": {
                    "valueSet": [
                        1,
                        2,
                        3,
                        4,
                        5,
                        6,
                        7,
                        8,
                        9,
                        10,
                        11,
                        12,
                        13,
                        14,
                        15,
                        16,
                        17,
                        18,
                        19,
                        20,
                        21,
                        22,
                        23,
                        24,
                        25
                    ]
                }
            },
            "hover": {
                "dynamicRegistration": true,
                "contentFormat": [
                    "markdown",
                    "plaintext"
                ]
            },
            "signatureHelp": {
                "dynamicRegistration": true,
                "signatureInformation": {
                    "documentationFormat": [
                        "markdown",
                        "plaintext"
                    ],
                    "parameterInformation": {
                        "labelOffsetSupport": true
                    },
                    "activeParameterSupport": true
                },
                "contextSupport": true
            },
            "definition": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "references": {
                "dynamicRegistration": true
            },
            "documentHighlight": {
                "dynamicRegistration": true
            },
            "documentSymbol": {
                "dynamicRegistration": true,
                "symbolKind": {
                    "valueSet": [
                        1,
                        2,
                        3,
                        4,
                        5,
                        6,
                        7,
                        8,
                        9,
                        10,
                        11,
                        12,
                        13,
                        14,
                        15,
                        16,
                        17,
                        18,
                        19,
                        20,
                        21,
                        22,
                        23,
                        24,
                        25,
                        26
                    ]
                },
                "hierarchicalDocumentSymbolSupport": true,
                "tagSupport": {
                    "valueSet": [
                        1
                    ]
                },
                "labelSupport": true
            },
            "codeAction": {
                "dynamicRegistration": true,
                "isPreferredSupport": true,
                "disabledSupport": true,
                "dataSupport": true,
                "resolveSupport": {
                    "properties": [
                        "edit"
                    ]
                },
                "codeActionLiteralSupport": {
                    "codeActionKind": {
                        "valueSet": [
                            "",
                            "quickfix",
                            "refactor",
                            "refactor.extract",
                            "refactor.inline",
                            "refactor.rewrite",
                            "source",
                            "source.organizeImports"
                        ]
                    }
                },
                "honorsChangeAnnotations": false
            },
            "codeLens": {
                "dynamicRegistration": true
            },
            "formatting": {
                "dynamicRegistration": true
            },
            "rangeFormatting": {
                "dynamicRegistration": true
            },
            "onTypeFormatting": {
                "dynamicRegistration": true
            },
            "rename": {
                "dynamicRegistration": true,
                "prepareSupport": true,
                "prepareSupportDefaultBehavior": 1,
                "honorsChangeAnnotations": true
            },
            "documentLink": {
                "dynamicRegistration": true,
                "tooltipSupport": true
            },
            "typeDefinition": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "implementation": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "colorProvider": {
                "dynamicRegistration": true
            },
            "foldingRange": {
                "dynamicRegistration": true,
                "rangeLimit": 5000,
                "lineFoldingOnly": true
            },
            "declaration": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "selectionRange": {
                "dynamicRegistration": true
            },
            "callHierarchy": {
                "dynamicRegistration": true
            },
            "semanticTokens": {
                "dynamicRegistration": true,
                "tokenTypes": [
                    "namespace",
                    "type",
                    "class",
                    "enum",
                    "interface",
                    "struct",
                    "typeParameter",
                    "parameter",
                    "variable",
                    "property",
                    "enumMember",
                    "event",
                    "function",
                    "method",
                    "macro",
                    "keyword",
                    "modifier",
                    "comment",
                    "string",
                    "number",
                    "regexp",
                    "operator"
                ],
                "tokenModifiers": [
                    "declaration",
                    "definition",
                    "readonly",
                    "static",
                    "deprecated",
                    "abstract",
                    "async",
                    "modification",
                    "documentation",
                    "defaultLibrary"
                ],
                "formats": [
                    "relative"
                ],
                "requests": {
                    "range": true,
                    "full": {
                        "delta": true
                    }
                },
                "multilineTokenSupport": false,
                "overlappingTokenSupport": false
            },
            "linkedEditingRange": {
                "dynamicRegistration": true
            }
        },
        "window": {
            "showMessage": {
                "messageActionItem": {
                    "additionalPropertiesSupport": true
                }
            },
            "showDocument": {
                "support": true
            },
            "workDoneProgress": true
        },
        "general": {
            "regularExpressions": {
                "engine": "ECMAScript",
                "version": "ES2020"
            },
            "markdown": {
                "parser": "marked",
                "version": "1.1.0"
            }
        }
    },
    "trace": "verbose",
    "workspaceFolders": [
        {
            "uri": "file:///c%3A/Users/Davy/swat.engineering/rascal/rascal-vscode/rascal-lsp",
            "name": "rascal-lsp"
        },
        {
            "uri": "file:///c%3A/Users/Davy/swat.engineering/rascal/rascal-vscode/rascal-vscode-extension",
            "name": "rascal-vscode-extension"
        },
        {
            "uri": "file:///c%3A/Users/Davy/swat.engineering/rascal/typepal",
            "name": "typepal"
        }
    ],
    "workDoneToken": "11162500-4a72-4009-b7e6-890878beacb6"
}
..
[Trace - 21:18:22] Received response 'initialize - (0)' in 1416ms.
Result: {
    "capabilities": {
        "textDocumentSync": 1,
        "definitionProvider": true,
        "documentSymbolProvider": true,
        "semanticTokensProvider": {
            "legend": {
                "tokenTypes": [
                    "entity.name",
                    "entity.other.inherited-class",
                    "entity.name.section",
.....
                    "variable.other.member",
                    "variable.other.readwrite",
                    "variable.other",
                    "variable.parameter"
                ],
                "tokenModifiers": []
            },
            "full": true
        }
    }
}


[Trace - 21:18:22] Sending notification 'initialized'.
Params: {}


[Trace - 21:18:22] Sending notification 'textDocument/didOpen'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Davy/swat.engineering/rascal/typepal/src/analysis/typepal/TModel.rsc",
        "languageId": "rascalmpl",
        "version": 1,
        "text": .... 
    }
}


[Trace - 21:18:22] Sending request 'textDocument/documentSymbol - (1)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Davy/swat.engineering/rascal/typepal/src/analysis/typepal/TModel.rsc"
    }
}


[Trace - 21:18:23] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Davy/swat.engineering/rascal/typepal/src/analysis/typepal/TModel.rsc",
    "diagnostics": []
}


[Trace - 21:18:33] Received response 'textDocument/documentSymbol - (1)' in 10434ms.
Result: [
....
]
```

# 2nd level

```
[Trace - 20:12:09] Sending request 'initialize - (0)'.
Params: {
    "processId": 9056,
    "clientInfo": {
        "name": "Visual Studio Code",
        "version": "1.54.1"
    },
    "locale": "en-gb",
    "rootPath": "c:\\Users\\Davy\\swat.engineering\\rascal\\rascal-lsp\\test project",
    "rootUri": "file:///c%3A/Users/Davy/swat.engineering/rascal/rascal-lsp/test%20project",
    "capabilities": {
        "workspace": {
            "applyEdit": true,
            "workspaceEdit": {
                "documentChanges": true,
                "resourceOperations": [
                    "create",
                    "rename",
                    "delete"
                ],
                "failureHandling": "textOnlyTransactional",
                "normalizesLineEndings": true,
                "changeAnnotationSupport": {
                    "groupsOnLabel": true
                }
            },
            "didChangeConfiguration": {
                "dynamicRegistration": true
            },
            "didChangeWatchedFiles": {
                "dynamicRegistration": true
            },
            "symbol": {
                "dynamicRegistration": true,
                "symbolKind": {
                    "valueSet": [
                        1,
                        2,
                        3,
                        4,
                        5,
                        6,
                        7,
                        8,
                        9,
                        10,
                        11,
                        12,
                        13,
                        14,
                        15,
                        16,
                        17,
                        18,
                        19,
                        20,
                        21,
                        22,
                        23,
                        24,
                        25,
                        26
                    ]
                },
                "tagSupport": {
                    "valueSet": [
                        1
                    ]
                }
            },
            "codeLens": {
                "refreshSupport": true
            },
            "executeCommand": {
                "dynamicRegistration": true
            },
            "configuration": true,
            "workspaceFolders": true,
            "semanticTokens": {
                "refreshSupport": true
            },
            "fileOperations": {
                "dynamicRegistration": true,
                "didCreate": true,
                "didRename": true,
                "didDelete": true,
                "willCreate": true,
                "willRename": true,
                "willDelete": true
            }
        },
        "textDocument": {
            "publishDiagnostics": {
                "relatedInformation": true,
                "versionSupport": false,
                "tagSupport": {
                    "valueSet": [
                        1,
                        2
                    ]
                },
                "codeDescriptionSupport": true,
                "dataSupport": true
            },
            "synchronization": {
                "dynamicRegistration": true,
                "willSave": true,
                "willSaveWaitUntil": true,
                "didSave": true
            },
            "completion": {
                "dynamicRegistration": true,
                "contextSupport": true,
                "completionItem": {
                    "snippetSupport": true,
                    "commitCharactersSupport": true,
                    "documentationFormat": [
                        "markdown",
                        "plaintext"
                    ],
                    "deprecatedSupport": true,
                    "preselectSupport": true,
                    "tagSupport": {
                        "valueSet": [
                            1
                        ]
                    },
                    "insertReplaceSupport": true,
                    "resolveSupport": {
                        "properties": [
                            "documentation",
                            "detail",
                            "additionalTextEdits"
                        ]
                    },
                    "insertTextModeSupport": {
                        "valueSet": [
                            1,
                            2
                        ]
                    }
                },
                "completionItemKind": {
                    "valueSet": [
                        1,
                        2,
                        3,
                        4,
                        5,
                        6,
                        7,
                        8,
                        9,
                        10,
                        11,
                        12,
                        13,
                        14,
                        15,
                        16,
                        17,
                        18,
                        19,
                        20,
                        21,
                        22,
                        23,
                        24,
                        25
                    ]
                }
            },
            "hover": {
                "dynamicRegistration": true,
                "contentFormat": [
                    "markdown",
                    "plaintext"
                ]
            },
            "signatureHelp": {
                "dynamicRegistration": true,
                "signatureInformation": {
                    "documentationFormat": [
                        "markdown",
                        "plaintext"
                    ],
                    "parameterInformation": {
                        "labelOffsetSupport": true
                    },
                    "activeParameterSupport": true
                },
                "contextSupport": true
            },
            "definition": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "references": {
                "dynamicRegistration": true
            },
            "documentHighlight": {
                "dynamicRegistration": true
            },
            "documentSymbol": {
                "dynamicRegistration": true,
                "symbolKind": {
                    "valueSet": [
                        1,
                        2,
                        3,
                        4,
                        5,
                        6,
                        7,
                        8,
                        9,
                        10,
                        11,
                        12,
                        13,
                        14,
                        15,
                        16,
                        17,
                        18,
                        19,
                        20,
                        21,
                        22,
                        23,
                        24,
                        25,
                        26
                    ]
                },
                "hierarchicalDocumentSymbolSupport": true,
                "tagSupport": {
                    "valueSet": [
                        1
                    ]
                },
                "labelSupport": true
            },
            "codeAction": {
                "dynamicRegistration": true,
                "isPreferredSupport": true,
                "disabledSupport": true,
                "dataSupport": true,
                "resolveSupport": {
                    "properties": [
                        "edit"
                    ]
                },
                "codeActionLiteralSupport": {
                    "codeActionKind": {
                        "valueSet": [
                            "",
                            "quickfix",
                            "refactor",
                            "refactor.extract",
                            "refactor.inline",
                            "refactor.rewrite",
                            "source",
                            "source.organizeImports"
                        ]
                    }
                },
                "honorsChangeAnnotations": false
            },
            "codeLens": {
                "dynamicRegistration": true
            },
            "formatting": {
                "dynamicRegistration": true
            },
            "rangeFormatting": {
                "dynamicRegistration": true
            },
            "onTypeFormatting": {
                "dynamicRegistration": true
            },
            "rename": {
                "dynamicRegistration": true,
                "prepareSupport": true,
                "prepareSupportDefaultBehavior": 1,
                "honorsChangeAnnotations": true
            },
            "documentLink": {
                "dynamicRegistration": true,
                "tooltipSupport": true
            },
            "typeDefinition": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "implementation": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "colorProvider": {
                "dynamicRegistration": true
            },
            "foldingRange": {
                "dynamicRegistration": true,
                "rangeLimit": 5000,
                "lineFoldingOnly": true
            },
            "declaration": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "selectionRange": {
                "dynamicRegistration": true
            },
            "callHierarchy": {
                "dynamicRegistration": true
            },
            "semanticTokens": {
                "dynamicRegistration": true,
                "tokenTypes": [
                    "namespace",
                    "type",
                    "class",
                    "enum",
                    "interface",
                    "struct",
                    "typeParameter",
                    "parameter",
                    "variable",
                    "property",
                    "enumMember",
                    "event",
                    "function",
                    "method",
                    "macro",
                    "keyword",
                    "modifier",
                    "comment",
                    "string",
                    "number",
                    "regexp",
                    "operator"
                ],
                "tokenModifiers": [
                    "declaration",
                    "definition",
                    "readonly",
                    "static",
                    "deprecated",
                    "abstract",
                    "async",
                    "modification",
                    "documentation",
                    "defaultLibrary"
                ],
                "formats": [
                    "relative"
                ],
                "requests": {
                    "range": true,
                    "full": {
                        "delta": true
                    }
                },
                "multilineTokenSupport": false,
                "overlappingTokenSupport": false
            },
            "linkedEditingRange": {
                "dynamicRegistration": true
            }
        },
        "window": {
            "showMessage": {
                "messageActionItem": {
                    "additionalPropertiesSupport": true
                }
            },
            "showDocument": {
                "support": true
            },
            "workDoneProgress": true
        },
        "general": {
            "regularExpressions": {
                "engine": "ECMAScript",
                "version": "ES2020"
            },
            "markdown": {
                "parser": "marked",
                "version": "1.1.0"
            }
        }
    },
    "trace": "verbose",
    "workspaceFolders": [
        {
            "uri": "file:///c%3A/Users/Davy/swat.engineering/rascal/rascal-lsp/test%20project",
            "name": "test project"
        }
    ],
    "workDoneToken": "05c1af7e-d626-45db-83a4-15406d00c9cb"
}

[Trace - 20:12:11] Received response 'initialize - (0)' in 1672ms.
Result: {
    "capabilities": {
        "textDocumentSync": 1,
        "definitionProvider": true,
        "documentSymbolProvider": true,
        "semanticTokensProvider": {
            "legend": {
                "tokenTypes": [
                    "entity.name",
                    "entity.other.inherited-class",
                    "entity.name.section",
                    "entity.name.tag",
                    "entity.other.attribute-name",
                    "variable",
....
                    "variable.annotation",
                    "variable.function",
                    "variable.language",
                    "variable.other.constant",
                    "variable.other.member",
                    "variable.other.readwrite",
                    "variable.other",
                    "variable.parameter"
                ],
                "tokenModifiers": []
            },
            "full": true
        }
    }
}


[Trace - 20:12:11] Sending notification 'initialized'.
Params: {}


[Trace - 20:12:11] Sending notification 'textDocument/didOpen'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Davy/swat.engineering/rascal/rascal-lsp/test%20project/src/Test1.rsc",
        "languageId": "rascalmpl",
        "version": 1,
        "text": "..."
    }
}


[Trace - 20:12:11] Sending request 'textDocument/documentSymbol - (1)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Davy/swat.engineering/rascal/rascal-lsp/test%20project/src/Test1.rsc"
    }
}


[Trace - 20:12:11] Sending request 'textDocument/semanticTokens/full - (2)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Davy/swat.engineering/rascal/rascal-lsp/test%20project/src/Test1.rsc"
    }
}


[Trace - 20:12:11] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Davy/swat.engineering/rascal/rascal-lsp/test%20project/src/Test1.rsc",
    "diagnostics": []
}


[Trace - 20:12:11] Received response 'textDocument/semanticTokens/full - (2)' in 214ms.
Result: {
    "data": [
        0,
        0,
        6,
        92,
        0,
.....
        92,
        0,
        1,
        4,
        6,
        92,
        0
    ]
}


[Trace - 20:12:21] Received response 'textDocument/documentSymbol - (1)' in 10567ms.
Result: [
...
]
```

In the trace logging I only see the difference that 2nd level we get the call for `semanticTokens/full` and in 1st we don't.

**Comment by DavyLandman** [comment]
Hmm, the only difference I can observe in the initialize request is that 2nd level has 
```
"linkedEditingRange": {
                "dynamicRegistration": true
            }
```
for the `"semanticTokens"` and 1st is lacking that option.

**Comment by dbaeumer** [comment]
The traces do look ok to me. Is there anything in the developer console (Help -> Toggle Developer tools) when running first level?

**Comment by DavyLandman** [comment]
Nope, nothing, I closed all rascal files, reloaded the window, waited till all extensions finished loading. Cleared the console, and only then opened a rascal file. I saw the LSP starting up, but the chrome console remained empty.

**Comment by dbaeumer** [comment]
Can you try to do the following:

- create a version of your extension first level that is not package / minified. 
- install that extension 
- close VS Code
- start VS Code from a terminal and use the `--inspect-extensions <port>        Allow debugging and profiling of extensions. Check the developer tools for the connection URI.` option
- connect a debugger (can be either VS Code or a Chrome debugger)
- can you try to see if the code is hit at the following locations after opening a rascal file:
https://github.com/microsoft/vscode-languageserver-node/blob/dbaeumer/revisedPull/client/src/common/semanticTokens.ts#L59
https://github.com/microsoft/vscode-languageserver-node/blob/dbaeumer/revisedPull/client/src/common/semanticTokens.ts#L110
https://github.com/microsoft/vscode-languageserver-node/blob/dbaeumer/revisedPull/client/src/common/semanticTokens.ts#L124

**Comment by DavyLandman** [comment]
Okay, I'll try next week.

Otherwise I'll try to create a small repo case that doesn't involve our whole plugin.

**Comment by dbaeumer** [comment]
Makes sense.

**Comment by DavyLandman** [comment]
Now I'm truly confused. I just installed the same package on a different windows machine (so same OS) and it works there. 

So something on my up to date version of vs code is not working nicely with Semantic Parsing, but not the other features? (and only in 1st level).

**Comment by DavyLandman** [comment]
I just removed the `~/.vscode/extensions/` folder and installed only our rascal extension, and the semantic highlighting works.

These are my installed extensions, I'll try to do some delta debugging to see if I can find a culperate.

```
asciidoctor.asciidoctor-vscode
bbrakenhoff.solarized-light-custom
Braver.vscode-solarized
brofox86.theme-espresso-soda-solarized
budparr.language-hugo-vscode
bungcip.better-toml
codezombiech.gitignore
DavidAnson.vscode-markdownlint
dbaeumer.vscode-eslint
donjayamanne.githistory
DotJoshJohnson.xml
eamodio.gitlens
eamodio.tsl-problem-matcher
ecmel.vscode-html-css
EditorConfig.EditorConfig
efbenson.scad
eg2.tslint
eg2.vscode-npm-script
faustinoaq.lex-flex-yacc-bison
felipecaputo.git-project-manager
gerane.Theme-Solarized-light
golang.go
grammarcraft.xtext-lang
hashhar.gitattributes
James-Yu.latex-workshop
jebbs.plantuml
joaompinto.asciidoctor-vscode
jumpinjackie.vscode-map-preview
mechatroner.rainbow-csv
mhmadhamster.postcss-language
ms-azuretools.vscode-docker
ms-python.python
ms-toolsai.jupyter
ms-vscode-remote.remote-wsl
ms-vscode.cpptools
mtxr.sqltools
nobuhito.printcode
PKief.material-icon-theme
platformio.platformio-ide
redhat.fabric8-analytics
redhat.java
redhat.vscode-commons
ritwickdey.LiveServer
ryanolsonx.solarized
scala-lang.scala
SonarSource.sonarlint-vscode
Strate.theme-solarized-light-strate
streetsidesoftware.code-spell-checker
timonwong.shellcheck
Togusa09.tmlanguage
usethesource.rascalmpl
VisualStudioExptTeam.vscodeintellicode
vscjava.vscode-java-debug
vscjava.vscode-java-dependency
vscjava.vscode-java-pack
vscjava.vscode-java-test
vscjava.vscode-maven
vscode-icons-team.vscode-icons
vscodevim.vim
Zignd.html-css-class-completion
```

**Comment by dbaeumer** [comment]
@aeschli can you clarify the rational behind this?

**Comment by DavyLandman** [comment]
(if you prefer, I can change the description of the issue to start a new one. As in the end it wasn't about deployment or 1st vs 2nd level. I disabled all extensions in my second level run, and that also disabled the theme extension).

**Comment by dbaeumer** [comment]
@DavyLandman you can leave as is for now.

**Comment by aeschli** [comment]
When we first added the semantic highlighting feature, there were strong reactions from users and theme authors to not interfere with the coloring of their carefully defined existing color themes. So we decided that we leave it to each theme to opt in to semantic highlighting. 
There's a setting`"editor.semanticHighlighting.enabled": true|false|"configuredByTheme"` with 'configuredByTheme' as the  default.

However, then language extension authors reached out to us. If a language heavily relies on semantic highlighting, then users will not get an coloring for themes that have not yet opted in.
As a solution to that, such a language can override the default of  `editor.semanticHighlighting.enabled`:
```json
    "configurationDefaults": {
      "[languageId]": {
         "editor.semanticHighlighting.enabled": true
      }
    }
```
